optimizely/optimizely.py

```diff
@@ -183,17 +187,30 @@ def _send_impression_event(self, project_config, experiment, variation, flag_key
                                user_id, attributes):
         """ Helper method to send impression event.
 
-    Args:
-      project_config: Instance of ProjectConfig.
-      experiment: Experiment for which impression event is being sent.
-      variation: Variation picked for user for the given experiment.
-      flag_key: key for a feature flag.
-      rule_key: key for an experiment.
-      rule_type: type for the source.
-      enabled: boolean representing if feature is enabled
-      user_id: ID for user.
-      attributes: Dict representing user attributes and values which need to be recorded.
-    """
+        Args:
+          project_config: Instance of ProjectConfig.
+          experiment: Experiment for which impression event is being sent.
+          variation: Variation picked for user for the given experiment.
+          flag_key: key for a feature flag.
+          rule_key: key for an experiment.
+          rule_type: type for the source.
+          enabled: boolean representing if feature is enabled
+          user_id: ID for user.
+          attributes: Dict representing user attributes and values which need to be recorded.
+        """
+        if not experiment:
```

Comment:
I will recommend to add a static method in Experiment class to get the default value, don't put empty in the impression event.

Reply:
@msohailhussain  Can you say more? A method that will get the default Experiment class, how will we make sure that class is empty?

Reply:
@msohailhussain I may need a bit of help with this one. Still working on it.

Reply:
Done.